src/hal/src/device.rs

```diff
@@ -734,6 +734,38 @@ pub trait Device<B: Backend>: fmt::Debug + Any + Send + Sync {
         flags: query::ResultFlags,
     ) -> Result<bool, WaitError>;
 
+    /// Create an acceleration structure object.
```

Comment:
Open question: does gfx have a plan on how to handle API extensions?

I currently plan on implementing this like the mesh shaders are implemented (i.e. have an implicit requirement that the `Feature` flag is available to be able to successfully call relevant methods). In my prototyping I realize this could become a maintenance burden (e.g. all non-supported backends need to stub out the methods and it's not extremely obvious which types and functions on device/commandlist are relevant to which extensions).

I can see a possibility with the extension trait pattern (unsupported backends would still need to stub out implementations, but it'd be very clear that e.g. ray tracing is not supported), or have the device/command list have a function to optionally return an extension-specific object with the relevant functions on it (unsupported backends can just return `None`, but I suspect implementing/using an API like this will be more cumbersome).

Reply:
Having traits would seem like a very nice quality of life improvement, but in practice for anything that tries to abstract over gfx-rs backends (i.e. `wgpu-core`) it would be a huge pain. Run-time flags are just more versatile and predictable for such a low-level library.

We could have the default implementations (doing `unimplemented!()`) to save on the boilerplate.

Reply:
Sounds good. I'll try and keep the `wgpu-core` use-case you mentioned in mind